### PR TITLE
Created AuthorizeRequest class for Paysafecard

### PR DESCRIPTION
**Pull request by ybenzine**
- Implemented `Paysafecard\AuthorizeRequest` class
- Added unit tests for public methods
- Removed an extra variable in `Astropay\AuthorizeRequestTest` class
- Ticket: https://webdmuk.atlassian.net/browse/VSP-1262

### CODE REVIEW

File: src/Message/Paysafecard/AuthorizeRequest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+namespace Omnipay\Noire\Message\Paysafecard;
+
+use Omnipay\Common\Exception\InvalidRequestException;
+use Omnipay\Noire\Helpers\PaymentMethodCodesHelperTrait;
+use Omnipay\Noire\Message\Paysafecard\AuthorizeResponse;
+use Omnipay\Noire\Message\OppwaBaseRequest;
+
+class AuthorizeRequest extends OppwaBaseRequest
+{
+    use PaymentMethodCodesHelperTrait;
```
Comment (Nicole8164) [inline src/Message/Paysafecard/AuthorizeRequest.php]: I don't believe this trait is needed for this file. It was for Astropay please remove

File: tests/Message/Astropay/AuthorizeRequestTest.php [hunk 2]
```diff
@@ -53,7 +53,6 @@ public function testGetData()
             'customer.surname' => 'Jones',
             'customer.identificationDocType' => 'TAXSTATEMENT',
             'paymentType' => 'DB',
-            'currency' => 'BRL',
```
Comment (Nicole8164) [inline tests/Message/Astropay/AuthorizeRequestTest.php]: Why was this removed?

File: tests/Message/Paysafecard/AuthorizeRequestTest.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    public function setUp()
+    {
+        $this->request = new AuthorizeRequest(
+            $this->getHttpClient(),
+            $this->getHttpRequest()
+        );
+
+        $this->request->initialize([
+            'channel' => '8a8294174b7ecb28014b9699220015ca',
```
Comment (Nicole8164) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: Is than actual channel and username?

File: tests/Message/Astropay/AuthorizeRequestTest.php (hunk 2 above)
Comment (ybenzine) [inline tests/Message/Astropay/AuthorizeRequestTest.php]: Because it's already defined [on line 45](https://github.com/vivastreet/omnipay-noire/pull/36/files/d4ad2de19ed50209e39422784e1b19dae22282c9#diff-84dec405df4f85db3df196b10bb37bd8L45) 😄

File: tests/Message/Paysafecard/AuthorizeRequestTest.php (hunk 3 above)
Comment (ybenzine) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: It's what they have on the test page, yes. I can replace it with dummy data though!

File: tests/Message/Paysafecard/AuthorizeRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public function testSetShopperResultUrl()
+    {
+        $expected_url = 'some url';
+
+        $this->request->setShopperResultUrl($expected_url);
+
+        $this->assertEquals($expected_url, $this->request->getShopperResultUrl());
+    }
+
+    public function testGetData()
```
Comment (jpcfernandes) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: This test description is unclear. If it returns whatever was passed in on initialisation, then it should be "testGetDataReturnsInitialisationData". In that case, ideally, the initialisation step should be on the test, not in the setup. When some initialisation is required, but specific values are not relevant to the test as long as things are valid, then it should go on the setUp/before part. If, on the other hand, the specific value is relevant to the test, it should be setup at the test.

File: tests/Message/Paysafecard/AuthorizeRequestTest.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        $expected_results = [
+            'authentication.entityId' => 'channel',
```
Comment (mcinbell) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: I'd suggest using class variable to tie the expected values with the actual values, i.e. the expected 'authentication.entityId' value is 'channel' because the request has been initialised with a 'channel' value of 'channel'. It helps the descriptiveness of unit tests to be explicit about where expected values come from.

File: tests/Message/Paysafecard/AuthorizeRequestTest.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
+<?php
+namespace Omnipay\Noire\Message\Paysafecard;
+
+use Guzzle\Http\ClientInterface;
+use Mockery as M;
+use Omnipay\Tests\TestCase;
+use Symfony\Component\HttpFoundation\Request as HttpRequest;
+
+class AuthorizeRequestTest extends TestCase
```
Comment (angelxmoreno) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: Seeing how `Omnipay\Noire\Message\Paysafecard\AuthorizeRequest` extends `OppwaBaseRequest` , is there any value from having `Omnipay\Noire\Message\Paysafecard\AuthorizeRequestTest` extend from a common TestClass so that only the differences in the various `AuthorizeRequest`s are tested? All `AuthorizeRequestTest` have common properties and follow an expected API. 

Not sure if the suggestion makes sense but would love to hear your thoughts.

Comment (ybenzine) [inline tests/Message/Paysafecard/AuthorizeRequestTest.php]: It actually does extend from an abstract `TestCase` class for shared/common functionality... 🙂 

Moreover, I checked both implementations of `AuthorizeRequestTest` (for both Paysafecard and Astropay) and while they have a bit of common logic, such as the way that they test the `getData` method, the input and output expected are different.

<@mcinbell's voice>There is a bit of room to DRY things up though</@mcinbell's voice> so I can try to make the `setUp` / `testGetData` methods a bit more abstract.
